src/us/kbase/workspace/database/provenance/PermanentID.java

```diff
@@ -0,0 +1,120 @@
+package us.kbase.workspace.database.provenance;
+
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * Represents a permanent unique identifier for an entity.
+ * Used for the 'related_identifiers' field of a {@link Resource} object.
```

Comment:
The only place in which this class is used -- 🤞  that makes the confusing documentation a bit less confusing.